backend/flashfolio/deck.go

```diff
@@ -157,6 +156,59 @@ func CloneDeckReq(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(ret)
 }
 
+func DeleteDeckReq(w http.ResponseWriter, r *http.Request) {
+
+	reqBody, err := ioutil.ReadAll(r.Body)
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	var req struct {
+		Deck  Deck   `json:"Deck"`
+		Token string `json:"Token"`
+	}
+
+	json.Unmarshal(reqBody, &req)
+
+	tokenInfo, err := VerifyIdToken(req.Token)
+	if err != nil {
+		/* Bad Token => Unauthorized */
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	user, err := GetUserByID(tokenInfo.UserId, ctx)
+
+	var savedDeck Deck
+	err = DeckCollection.FindOne(ctx, bson.D{{Key: "id", Value: req.Deck.ID}}).Decode(&savedDeck)
+	if err != nil {
+		/* That deck isn't saved to the database*/
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+	DeckCollection.DeleteOne(ctx, bson.D{{Key: "id", Value: req.Deck.ID}})
```

Comment:
Please make a check to see if the user making the request matches the owner of the deck before deleting in. Right now a check is only made to see if the token is valid. This means that a user who doesn't own a deck could remove the deck of a different user.

Exempli Gratia:

1. User A creates deck 9001.
2. User B sends a request to the back end to delete deck 9001, using their valid token.
3. Backend checks to see if the token is valid, and it is.
4. The backend deletes deck 9001.

Ergo: User B has deleted User A's deck, despite not being the owner. A check should be made between 3 & 4 to see if the deck owner matches the token owner.

Reply:
Done.